### PR TITLE
Fix: initialize batch norm in YOLOX

### PR DESCRIPTION
**Pull request by liyier90**
- Initialize `nn.BatchNorm2d` with the same parameters as the original repo
- Add class aware NMS (`torchvision.ops.batched_nms`) to match benchmarking config of original repo
- Update tests with new groundtruths
- Remove empty `detect_ids` list behavior, made redundant by [this update](https://github.com/aimakerspace/PeekingDuck/blob/docs-v1.2/peekingduck/utils/create_node_helper.py#L199-L201)

### CODE REVIEW

**Comment by leeping-ng** [comment]
LGTM
